core/src/main/scala/cats/instances/equiv.scala

```diff
@@ -22,8 +22,51 @@
 package cats
 package instances
 
-trait EquivInstances {
-  implicit val catsContravariantMonoidalForEquiv: ContravariantMonoidal[Equiv] =
+trait EquivInstances extends EquivInstances0 {
+  implicit def catsDecidableForEquiv: Decidable[Equiv] =
+    new Decidable[Equiv] {
+
+      /**
+       * Defaults to trivially contracting the type
+       * to a point
+       */
+      def unit: Equiv[Unit] =
+        new Equiv[Unit] {
+          def equiv(x: Unit, y: Unit): Boolean = true
+        }
+
+      /**
+       * Derive an `Equiv` for `B` given an `Equiv[A]` and a function `B => A`.
+       *
+       * Note: resulting instances are law-abiding only when the functions used are injective (represent a one-to-one mapping)
+       */
+      def contramap[A, B](fa: Equiv[A])(f: B => A): Equiv[B] =
+        new Equiv[B] {
+          def equiv(l: B, r: B): Boolean =
+            fa.equiv(f(l), f(r))
+        }
+
+      def product[A, B](fa: Equiv[A], fb: Equiv[B]): Equiv[(A, B)] =
+        new Equiv[(A, B)] {
+          def equiv(l: (A, B), r: (A, B)): Boolean =
+            fa.equiv(l._1, r._1) && fb.equiv(l._2, r._2)
+        }
+
+      def sum[A, B](fa: Equiv[A], fb: Equiv[B]): Equiv[Either[A, B]] =
+        new Equiv[Either[A, B]] {
+          def equiv(x: Either[A, B], y: Either[A, B]): Boolean = (x, y) match {
+            case (Left(a1), Left(a2))   => fa.equiv(a1, a2)
+            case (Right(b1), Right(b2)) => fb.equiv(b1, b2)
+            case _                      => false
+          }
+        }
+
+      override def zero: Equiv[Nothing] = Equiv.by[Nothing, Unit](_ => ())
```

Comment:
can make this a `val` to avoid reallocating...

Reply:
Good call, I'll give the `def`s a pass and make `val` where appropriate.